stats/cloud/collector.go

```diff
@@ -462,9 +490,12 @@ func (c *Collector) pushMetrics() {
 		}
 		err := c.client.PushMetric(c.referenceID, c.config.NoCompress.Bool, buffer[:size])
 		if err != nil {
-			logrus.WithFields(logrus.Fields{
-				"error": err,
-			}).Warn("Failed to send metrics to cloud")
+			if c.shouldStopSendingMetrics(err) {
+				logrus.Warn("Stopped sending metrics to cloud due to an error.")
```

Comment:
Missing `.WithError(err)` here. Also this warning message ends with a dot, while none of the others do.